lib/transport-common.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const Connection = require('./connection');
+const Application = require('./applications').Application;
+const SimpleConnectPolicy = require('./simple-connect-policy');
+const common = require('./common');
+
+// Create a function to create a JSTP client with connFactory and
+// transportFactory.
+//   connFactory - function that will be called with ...options
+//                 and must return rawConnection in callback in a form
+//                 (error, rawConnection)
+//   transportFactory - function that will be called with rawConnection
+//                      and must return transport
+//
+// returns function with arguments
+//   appName - remote application name to connect to
+//   client - client object, may be null, then it will be replaced with
+//            client with default application (see jstp.Application)
+//            if not null must contain 'application'
+//            (if it doesn't default application will be used)
+//            can contain 'connectPolicy' if there is no
+//            connectPolicy default SimpleConnectPolicy will be used
+//            if it contains heartbeatInterval then heartbeat on JSTP
+//            connection will be started with it
+//   options - will be destructured and passed directly to connFactory
+//             The last argument of options is optional callback
+//             that will be called when connection is established
+//
+const createConnectionFactory = (
+  connFactory, transportFactory
+) => (appName, client, ...options) => {
+  const callback = common.extractCallback(options);
+  connFactory(...options, (error, rawConnection) => {
+    if (error) return callback(error);
+
+    const transport = transportFactory(rawConnection);
+    if (!client) {
+      client = { application: new Application('jstp', {}) };
```

Comment:
Can you please add `, connectPolicy: null` here so that the object shape does not change at line 44?